### PR TITLE
feat(utxo-lib): parse psbt for p2trMusig2

### PR DESCRIPTION
**Pull request by saravanan7mani**
parsePubScript and parsePsbtInput now supports p2trMusig2

TICKET: BG-66942

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 1]
```diff
@@ -7,31 +8,58 @@ import { createPsbtFromTransaction } from '../transaction';
...
+}
 
-export interface ParsedPsbt2Of3 extends ParsedPubScriptP2ms {
-  signatures: Signatures;
+/**
+ * To hold parsed signature data for TaprootKeyPathSpend script type.
+ * signatures and participantPublicKeys (signer pub keys)
+ */
+export interface TaprootKeyPathSignatures {
+  signatures: [Buffer] | [Buffer, Buffer] | undefined;
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/wallet/Psbt.ts:36]: If I understand this correctly
```suggestion
  /** Two signatures for the pre-aggregated state */ 
  signatures: [Buffer] | [Buffer, Buffer] | undefined;
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
small nits

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 2]
```diff
@@ -7,31 +8,58 @@ import { createPsbtFromTransaction } from '../transaction';
...
+  ParsedPubScriptTaproot,
+  ParsedPubScriptTaprootKeyPath,
   ParsedScriptType2Of3,
 } from '../parseInput';
+import { parsePsbtMusig2PartialSigs } from '../Musig2';
+import { isTuple } from '../types';
+import { createTaprootOutputScript } from '../../taproot';
 
-type Signatures = [Buffer] | [Buffer, Buffer] | undefined;
+export interface Signatures {
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts]: Done

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts]: Done

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 3]
```diff
@@ -7,31 +8,58 @@ import { createPsbtFromTransaction } from '../transaction';
...
+  signatures: [Buffer] | [Buffer, Buffer] | undefined;
+}
 
-export interface ParsedPsbt2Of3 extends ParsedPubScriptP2ms {
-  signatures: Signatures;
+/**
+ * To hold parsed signature data for TaprootKeyPathSpend script type.
+ * signatures and participantPublicKeys (signer pub keys)
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts]: Done

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 4]
```diff
@@ -7,31 +8,58 @@ import { createPsbtFromTransaction } from '../transaction';
...
+ */
+export interface TaprootKeyPathSignatures {
+  signatures: [Buffer] | [Buffer, Buffer] | undefined;
+  // signer pub keys.
+  participantPublicKeys: [Buffer] | [Buffer, Buffer] | undefined;
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts]: Done

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**